src/utils/utils.ts

```diff
@@ -148,3 +150,27 @@ export const calculateFileSize = (sizeInBytes: number): string => {
 
   return (sizeInBytes / Math.pow(THRESHOLD, exp)).toFixed(DECIMAL_PLACES) + SEPARATOR + units[exp];
 };
+
+interface ICheckColumnsCombinations {
+  columns: Array<string>;
+  combinations: Array<Array<string>>;
+}
+
+/**
+ * Function to check if the combination requirements has been satisfied.
+ *
+ * @param columns - columns to be checked
+ * @param combinations - the combination requirements that need to be satisfied
+ *
+ */
+export const checkColumnsCombinations = ({ columns, combinations }: ICheckColumnsCombinations) => {
+  let isMismatch = false;
+  combinations.every((combination) => {
```

Comment:
if `true` and `false` are returned, then I would suggest following version should be used:

```tsx
return combinations.every((combination) => {
```

Then also `boolean` return type for the method needs to be added.

Reply:
The reason for using `every` instead of `forEach` is that `every` will terminate the iteration when first `false` is returned, and either `true` or `false` is required to be returned in each iteration. See doc: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

Reply:
Using `every` is fine, also `true` and `false`, but I was meaning also returning it would be more usual. If you prefer your void based version, let's keep it, we can change it later when required.

Reply:
Updated accordingly